### PR TITLE
DT-641 Basic no-op endpoint for matching

### PR DESCRIPTION
**Pull request by andymarke**
WIP to add new endpoint with validation of the request

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/advice/ControllerAdvice.kt [hunk 1]
```diff
@@ -77,4 +79,25 @@ class ControllerAdvice {
...
+    log.debug("Bad request (400) returned", e)
+    return ResponseEntity
+        .status(HttpStatus.BAD_REQUEST)
+        .body(ErrorResponse(status = HttpStatus.BAD_REQUEST.value(), developerMessage = e.developerMessage()))
+  }
+
+}
+
+private fun MethodArgumentNotValidException.developerMessage(): String {
+  return this.bindingResult.allErrors.joinToString { it.defaultMessage ?: "unknown" }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/advice/ControllerAdvice.kt]: default message was unreadable so added this

File: src/main/resources/logback-spring.xml [hunk 2]
```diff
@@ -48,7 +48,7 @@
         <appender-ref ref="consoleAppender"/>
     </logger>
 
-    <root level="DEBUG">
+    <root level="INFO">
```
Comment (petergphillips) [inline src/main/resources/logback-spring.xml]: is this deliberate - we normally run on `DEBUG` for all services

File: src/main/kotlin/uk/gov/justice/hmpps/offendersearch/dto/MatchRequest.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import io.swagger.annotations.ApiModelProperty
+import java.time.LocalDate
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.Past
+
+data class MatchRequest(
+    @ApiModelProperty(value = "Offender first name", example = "john",  position = 1) val firstName: String? = null,
+    @field:NotBlank(message = "Surname is required")
+    @ApiModelProperty(value = "Offender surname", example = "smith", position = 2) val surname: String? = null,
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/hmpps/offendersearch/dto/MatchRequest.kt]: do you need a default `null` value if the field is required

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/hmpps/offendersearch/dto/MatchRequest.kt]: agreed, I'll fix

File: src/main/resources/logback-spring.xml (hunk 2 above)
Comment (andymarke) [reply src/main/resources/logback-spring.xml]: All work why there was so much noise in DEBUG mode and correct

**Pull request merged**